linera-core/src/node.rs

```diff
@@ -103,6 +102,10 @@ pub trait ValidatorNode {
     /// Subscribes to receiving notifications for a collection of chains.
     async fn subscribe(&self, chains: Vec<ChainId>) -> Result<Self::NotificationStream, NodeError>;
 
+    // Uploads a blob content. Returns an error if the validator has not seen a
+    // certificate using this blob.
```

Comment:
Where do we do that? i.e. check that validator has not seen the certificate yet?

Reply:
In proxy/grpc.rs:485: If the validator hasn't seen it yet, there's no blob state, so it doesn't write the blob.